wemake_python_styleguide/violations/best_practices.py

```diff
@@ -1417,3 +1417,64 @@ class IncorrectSuperCallViolation(ASTViolation):
 
     error_template = 'Found incorrect `super()` call: {0}'
     code = 456
+
+
+@final
+class VariableUsedOutsideOfWithBlockViolation(ASTViolation):
```

Comment:
I guess that we can have a single violation class for both of this cases.
I would name it `VariableUsedOutsideOfBlockViolation`.

You also have to update the docs on top of this module to add a summary and `autoclass` directive.